neural.py

```diff
@@ -213,81 +216,82 @@ def get_prob_type(self):
         'continuous' or 'either'
         """
         return self.prob_type
-        
+
     def calculate_updates(self):
         """Calculate gradient descent updates.
-        
+
         Args:
         None
-        
+
         Returns:
         updates_list: list. List of back propagation weight updates.
         """
         delta_list = []
         updates_list = []
-        
+
         # Work backwards from final layer
         for i in range(len(self.inputs_list)-1, -1, -1):
             # Final layer
             if i == len(self.inputs_list)-1:
                 delta = (self.y_pred - self.y_true)
             # Hidden layers
             else:
-                delta = np.dot(delta_list[-1], 
-                               np.transpose(self.weights[i+1]))*\
-                               self.activation(self.inputs_list[i+1], 
-                                               deriv=True)
-            
+                delta = np.dot(delta_list[-1],
```

Comment:
I think the original is correct here, but the length of the expression may be affecting its understandability. 
What I am trying to do is `(1)` times `(2)` where:
`(1) = np.dot(delta_list[-1], np.transpose(self.weights[i+1]))`
and 
`(2) = self.activation(self.inputs_list[i+1], deriv=True)
`
Perhaps if we changed the formatting slightly to the following, it would be clearer:
```suggestion
                delta = np.dot(delta_list[-1], 
                               np.transpose(self.weights[i+1]))*\
                                  self.activation(self.inputs_list[i+1], 
                                                  deriv=True)
```

Reply:
With regard to your feedback on line 242 of neural.py, see the above review comment. `deriv=True` is a parameter of `self.activation`, not `np.dot`. However, I agree that the formatting should change to make this less ambiguous.

With regard to your side-note, that's a good suggestion. I will make that change.

Reply:
I think that formatting it differently will be a good move. I don't think that the line continuation ('\') is the best idea but I can look at how to best format.

Reply:
Thank you. That would be a great help.